### ISSUE TITLE
mongodb package compatibility

### ISSUE THREAD

**Issue by diegoaguilar** [balderdashy/sails-mongo#299]
As I understand, it's now required to get mongodb module installed in order to work with sails-mongo.
I got both mongodb and sails-mongo installed and when I try to run `sails lift` I will obtain the following log:

```
/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/mongodb/lib/url_parser.js:52
    auth[1] = decodeURIComponent(auth[1]);
              ^
URIError: URI malformed
    at decodeURIComponent (native)
    at module.exports (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/mongodb/lib/url_parser.js:52:15)
    at Function.MongoClient.connect (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/mongodb/lib/mongo_client.js:95:16)
    at Connection._buildConnection (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails-mongo/lib/connection.js:149:15)
    at new Connection (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails-mongo/lib/connection.js:24:8)
    at Object.module.exports.adapter.registerConnection (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails-mongo/lib/adapter.js:97:22)
    at /home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/lib/waterline.js:203:29
    at /home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:157:13
    at _each (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:57:9)
    at Object.async.each (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:156:9)
    at Array.async.auto.registerConnections [as 1] (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/lib/waterline.js:176:13)
    at /home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:640:38
    at _each (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:57:9)
    at Object.async.auto (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/node_modules/async/lib/async.js:600:9)
    at Waterline.initialize (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/waterline/lib/waterline.js:121:9)
    at buildORM (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/lib/hooks/orm/build-orm.js:52:15)
    at Array.async.auto.instantiatedCollections [as 1] (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/lib/hooks/orm/index.js:203:11)
    at listener (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/async/lib/async.js:490:46)
    at /home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/async/lib/async.js:441:17
    at _each (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/async/lib/async.js:46:13)
    at Object.taskComplete (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails/node_modules/async/lib/async.js:440:13)
    at processImmediate [as _immediateCallback] (timers.js:354:15)
Program node app.js exited with code 8
```
I was testing this against a 2.4 MongoDB database. At my package.json I got as dependencies `"sails-mongo": "^0.11.2"` and `"mongodb": "^2.0.34"`

**Comment by dominicrico** [comment]
Got the same error ...

**Comment by devinivy** [comment]
Sorry, I don't fully understand the issue here, and I'm a little confused!  Why doesn't sails-mongo use the version of mongodb that it has in its own package.json?

Do you understand what's happening between these two lines of the stacktrace, where sails-mysql uses its sibling dependency?  It seems odd.  Maybe I'm having a brain-cramp.
```
    at Function.MongoClient.connect (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/mongodb/lib/mongo_client.js:95:16)
    at Connection._buildConnection (/home/diegoaguilar/dev/homestead/data4/liga-fantastica/node_modules/sails-mongo/lib/connection.js:149:15)
```

**Comment by sailsbot** [comment]
Thanks for posting, @diegoaguilar.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
